src/testing/vortex/workload.zig

```diff
@@ -0,0 +1,432 @@
+            const transfers = command.create_transfers;
+
+            for (entries, 0..) |entry, e| {
+                const transfer = transfers[entry.index];
+
+                // Check that linked transfers fail together.
+                if (entry.index > 0) {
+                    const preceeding_transfer = transfers[entry.index - 1];
+                    if (preceeding_transfer.flags.linked) {
+                        try testing.expect(e > 0);
+                        const preceeding_entry = entries[e - 1];
+                        try testing.expect(preceeding_entry.index == entry.index - 1);
+                        try testing.expect(preceeding_entry.result != .ok);
+                    }
+                }
+
+                // No further validation needed for failed transfers.
+                if (entry.result != .ok) {
+                    continue;
+                }
+
+                if (transfer.flags.pending) {
+                    try testing.expect(!model.pending_transfers.contains(transfer.id));
+                    try model.pending_transfers.put(transfer.id, {});
+                    // We can't let this grow unboundedly:
+                    assert(model.pending_transfers.count() <= pending_transfers_count_max);
+                }
+
+                if (transfer.flags.void_pending_transfer or transfer.flags.post_pending_transfer) {
+                    try testing.expect(model.pending_transfers.remove(transfer.id));
+                }
+
+                try testing.expect(model.account_exists(transfer.debit_account_id));
+                try testing.expect(model.account_exists(transfer.credit_account_id));
+            }
+        },
+        .lookup_all_accounts => |accounts_found| {
+            // Get all known account ids (sorted).
+            var id_buffer: [accounts_count_max]u128 = undefined;
+            const account_ids_known = model.account_ids(id_buffer[0..]);
+
+            // Extract and sort all found account ids.
+            var account_ids_found_buffer: [accounts_count_max]u128 = undefined;
+            for (accounts_found, 0..) |account, i| {
+                account_ids_found_buffer[i] = account.id;
+            }
+            const account_ids_found = account_ids_found_buffer[0..accounts_found.len];
+            std.mem.sort(u128, account_ids_found, {}, std.sort.asc(u128));
+
+            // All known accounts are found by the query, and no others.
+            try testing.expectEqualSlices(u128, account_ids_known, account_ids_found);
+
+            try testing.expectEqual(0, debits_credits_difference(accounts_found));
+        },
+    }
+}
+
+/// Tracks information about the accounts and transfers created by the workload.
+const Model = struct {
+    accounts: std.ArrayListUnmanaged(tb.Account),
+    pending_transfers: std.AutoHashMap(u128, void),
+
+    // O(n) lookup, but it's limited by `accounts_count_max`, so it's OK for this test.
+    fn account_exists(self: @This(), id: u128) bool {
```

Comment:
Maybe model rather than self? We try to use self less, to keep the naming unfimorm between methods and functions.